### PR TITLE
Add CMake options EMBED_RESOURCE_FILES and USE_ONLY_EMBEDDED_RESOURCE_FILES for proj.db and proj.ini embedding

### PR DESCRIPTION
**Pull request by rouault**
Implements https://github.com/OSGeo/PROJ/pull/4274

### CODE REVIEW

File: cmake/FileEmbed.cmake [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}\;
+unsigned ${c_name}_size = sizeof(${c_name}_data)\;
+")
+
+    set(output_h "
+#ifndef ${c_name}_H
+#define ${c_name}_H
+#include \"stdint.h\"
+extern const uint8_t ${c_name}_data[]\;
+extern unsigned ${c_name}_size\;
```
Comment (eschnett) [inline cmake/FileEmbed.cmake]: Add `const`?

File: cmake/FileEmbed.cmake [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+const uint8_t ${c_name}_data[] = {
+    ${output_c}
+}\;
+unsigned ${c_name}_size = sizeof(${c_name}_data)\;
+")
+
+    set(output_h "
+#ifndef ${c_name}_H
+#define ${c_name}_H
+#include \"stdint.h\"
```
Comment (eschnett) [inline cmake/FileEmbed.cmake]: Use `#include <stdint.h>` instead? https://en.cppreference.com/w/c/preprocessor/include says that the `<...>` form is intended for standard include files.

File: cmake/FileEmbed.cmake [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    endif()
+
+    set(output_c "${SEPARATED_HEX}${SEPARATED_HEX_REMAINDER}")
+
+    set(output_c "
+#include \"${c_name}.h\"
+const uint8_t ${c_name}_data[] = {
+    ${output_c}
+}\;
+unsigned ${c_name}_size = sizeof(${c_name}_data)\;
```
Comment (eschnett) [inline cmake/FileEmbed.cmake]: Use `size_t` instead of `unsigned`?

File: cmake/FileEmbed.cmake (hunk 1 above)
Comment (eschnett) [inline cmake/FileEmbed.cmake]: In principle the `..._size` constant isn't needed. It isn't generated by the `#embed` directive either.

Comment (rouault) [reply cmake/FileEmbed.cmake]: done

File: cmake/FileEmbed.cmake (hunk 2 above)
Comment (rouault) [reply cmake/FileEmbed.cmake]: done

File: cmake/FileEmbed.cmake (hunk 3 above)
Comment (rouault) [reply cmake/FileEmbed.cmake]: we are not going to embed more than 2 GB, so unsigned is fine

File: cmake/FileEmbed.cmake (hunk 1 above)
Comment (rouault) [reply cmake/FileEmbed.cmake]: As we are accessing the resource through the .h file, we cannot do a sizeof(proj_db_data), so we need a variable giving the size. We do need the size to pass it to the SQLite3 memvfs

Comment (eschnett) [reply cmake/FileEmbed.cmake]: D'oh.

**Pull request merged**